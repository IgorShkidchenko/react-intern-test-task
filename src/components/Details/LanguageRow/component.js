import React from 'react'

const LanguageRowComponent = (props) => {
  const percentage = props.percentage
  const languageName = props.languageName

  return (
    <tr>
      <td>{languageName}: {percentage}%</td>
    </tr>
  )
}

export default LanguageRowComponent
