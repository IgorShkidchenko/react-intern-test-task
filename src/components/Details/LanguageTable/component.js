import React from 'react'

const LanguageTableComponent = ({ rows }) => {
  return (
    <table>
      <thead>
        <tr>
          <td>Languages:</td>
        </tr>
      </thead>
      <tbody>{rows}</tbody>
    </table>
  );
}

export default LanguageTableComponent
