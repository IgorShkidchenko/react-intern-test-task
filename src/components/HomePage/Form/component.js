import React from "react";
import PropTypes from "prop-types";

const FormComponent = ({
  handleSubmit, handleChange, value, valid,
}) => (
  <form onSubmit={handleSubmit}>
    <input type="text" placeholder="Enter repository name e.g. facebook/react" value={value} onChange={handleChange} />
    <input type="submit" value="Add" />
    {valid === false ? <p className="red">Repository not found</p> : null}
  </form>
);

FormComponent.propTypes = {
  value: PropTypes.string.isRequired,
  valid: PropTypes.bool.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default FormComponent;
