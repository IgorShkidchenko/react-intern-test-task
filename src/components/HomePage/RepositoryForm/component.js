import React from "react";
import PropTypes from "prop-types";
import { Form, Field, ErrorMessage } from "formik";

const RepositoryFormComponent = ({ isSubmitting }) => (
  <Form>
    <Field type="repositoryUrl" name="repositoryUrl" placeholder="Enter repository name e.g. facebook/react" />
    <ErrorMessage name="repositoryUrl" component="p" />
    <button type="submit" disabled={isSubmitting}>
      Submit
    </button>
  </Form>
);

RepositoryFormComponent.propTypes = {
  isSubmitting: PropTypes.bool.isRequired,
};

export default RepositoryFormComponent;
