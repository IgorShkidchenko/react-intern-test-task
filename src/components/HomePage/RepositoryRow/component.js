import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import * as path from "../../../constants/path";

const RepotisoryRowComponent = ({ repository, handleClick }) => (
  <tr>
    <td>
      <Link to={{
            pathname: path.DETAILS,
            state: {
              repository,
            },
          }}
      >
        {repository.name} {repository.stargazers_count}
        <span
          role="button"
          tabIndex={0}
          onClick={handleClick}
          onKeyDown={handleClick}
        >  Ð¥
        </span>
      </Link>
    </td>
  </tr>
);

RepotisoryRowComponent.propTypes = {
  repository: PropTypes.object.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default RepotisoryRowComponent;
