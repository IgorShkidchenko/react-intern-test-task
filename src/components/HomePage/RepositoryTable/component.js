import React from "react";
import PropTypes from "prop-types";

const RepositoryTableComponent = ({ rows }) => (
  <table>
    <thead>
      <tr>
        <th>Repository Name</th>
        <th>Stars</th>
      </tr>
    </thead>
    <tbody>{rows}</tbody>
  </table>
);

RepositoryTableComponent.propTypes = {
  rows: PropTypes.array.isRequired,
};

export default RepositoryTableComponent;
