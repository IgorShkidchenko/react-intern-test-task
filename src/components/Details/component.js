import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import * as path from "../../constants/path";

import Stats from "./Stats";
import LanguageTable from "./LanguageTable";

const DetailsComponent = ({ location }) => {
  const { repository } = location.state;

  return (
    <div>
      <p>
        {repository.full_name}
        <a href={repository.clone_url}> Clone</a>
      </p>
      <div className="tables">
        <Stats repository={repository} />
        <LanguageTable repository={repository} />
      </div>
      <Link to={path.ROOT}>Back</Link>
    </div>
  );
};

DetailsComponent.propTypes = {
  location: PropTypes.object.isRequired,
};

export default DetailsComponent;
