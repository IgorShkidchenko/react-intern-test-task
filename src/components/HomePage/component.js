import React from "react";

import RepositoryForm from "./RepositoryForm";
import RepositoryTable from "./RepositoryTable";

const HomePageComponent = () => (
  <div>
    <RepositoryForm />
    <RepositoryTable />
  </div>
);

export default HomePageComponent;
