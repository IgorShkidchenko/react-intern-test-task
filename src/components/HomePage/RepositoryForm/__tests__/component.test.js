import React from "react";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { waitFor } from "@testing-library/dom";
import { RepositoryForm } from "../component";

configure({ adapter: new Adapter() });

describe("<RepotisoryRowComponent />", () => {
  const clickFn = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<RepositoryForm
      onAddRepository={clickFn}
    />);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call onAddRepository on submit", async () => {
    const value = { target: { name: "repositoryUrl", value: "facebook/react" } };
    wrapper.find("input").simulate("change", value);

    await waitFor(() => {
      wrapper.find("form").simulate("submit");
      expect(clickFn).toHaveBeenCalledTimes(1);
    });
  });
});
