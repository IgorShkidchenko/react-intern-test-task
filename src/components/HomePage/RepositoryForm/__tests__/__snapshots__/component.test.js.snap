// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepotisoryRowComponent /> renders correctly 1`] = `
<RepositoryForm
  onAddRepository={[MockFunction]}
>
  <Formik
    initialValues={
      Object {
        "repositoryUrl": "",
      }
    }
    onSubmit={[Function]}
    validate={[Function]}
  >
    <Form>
      <form
        action="#"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <Field
          name="repositoryUrl"
          placeholder="Enter repository name e.g. facebook/react"
          type="repositoryUrl"
        >
          <input
            name="repositoryUrl"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter repository name e.g. facebook/react"
            type="repositoryUrl"
            value=""
          />
        </Field>
        <FormikConnect(ErrorMessageImpl)
          component="p"
          name="repositoryUrl"
        >
          <ErrorMessageImpl
            component="p"
            formik={
              Object {
                "dirty": false,
                "errors": Object {},
                "getFieldHelpers": [Function],
                "getFieldMeta": [Function],
                "getFieldProps": [Function],
                "handleBlur": [Function],
                "handleChange": [Function],
                "handleReset": [Function],
                "handleSubmit": [Function],
                "initialErrors": Object {},
                "initialStatus": undefined,
                "initialTouched": Object {},
                "initialValues": Object {
                  "repositoryUrl": "",
                },
                "isSubmitting": false,
                "isValid": true,
                "isValidating": false,
                "registerField": [Function],
                "resetForm": [Function],
                "setErrors": [Function],
                "setFieldError": [Function],
                "setFieldTouched": [Function],
                "setFieldValue": [Function],
                "setFormikState": [Function],
                "setStatus": [Function],
                "setSubmitting": [Function],
                "setTouched": [Function],
                "setValues": [Function],
                "status": undefined,
                "submitCount": 0,
                "submitForm": [Function],
                "touched": Object {},
                "unregisterField": [Function],
                "validateField": [Function],
                "validateForm": [Function],
                "validateOnBlur": true,
                "validateOnChange": true,
                "validateOnMount": false,
                "values": Object {
                  "repositoryUrl": "",
                },
              }
            }
            name="repositoryUrl"
          />
        </FormikConnect(ErrorMessageImpl)>
        <button
          disabled={false}
          type="submit"
        >
          Submit
        </button>
      </form>
    </Form>
  </Formik>
</RepositoryForm>
`;
